File: data/src/main/java/io/forus/me/android/data/entity/serializers/SignRecordsTypeAdapter.java
package io.forus.me.android.data.entity.serializers;

import com.fernandocejas.arrow.strings.Strings;
import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

import java.io.IOException;

import io.forus.me.android.data.entity.sign.request.SignRecords;

public class SignRecordsTypeAdapter extends TypeAdapter<SignRecords> {

    @Override
    public void write(JsonWriter out, SignRecords value) throws IOException {
        out.beginObject();
        if (!Strings.isNullOrEmpty(value.getEmail())) {
            out.name("primary_email");
            out.value(value.getEmail());
        }

        if (!Strings.isNullOrEmpty(value.getFirstname())) {
            out.name("given_name");
            out.value(value.getFirstname());
        }

        if (!Strings.isNullOrEmpty(value.getLastname())) {
            out.name("family_name");
            out.value(value.getLastname());
        }

        if (!Strings.isNullOrEmpty(value.getBsn())) {
            out.name("bsn");
            out.value(value.getBsn());
        }

        if (!Strings.isNullOrEmpty(value.getPhoneNumber())) {
            out.name("telephone");
            out.value(value.getPhoneNumber());
        }

        out.endObject();
    }

    @Override
    public SignRecords read(JsonReader in) {
        return null;
    }
}

File: data/src/main/java/io/forus/me/android/data/entity/serializers/SignRecordsTypeAdapter.kt
package io.forus.me.android.data.entity.serializers

import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter
import io.forus.me.android.data.entity.sign.request.SignRecords
import java.io.IOException

class SignRecordsTypeAdapter : TypeAdapter<SignRecords?>() {
    @Throws(IOException::class)
    override fun write(out: JsonWriter, value: SignRecords?) {
        out.beginObject()

        value?.firstname?.let {
            out.name("given_name")
            out.value(value.firstname)
        }

        value?.lastname?.let {
            out.name("family_name")
            out.value(value.lastname)
        }

        value?.bsn?.let {
            out.name("bsn")
            out.value(value.bsn)
        }

        value?.phoneNumber?.let {
            out.name("telephone")
            out.value(value.phoneNumber)
        }

        out.endObject()
    }

    override fun read(`in`: JsonReader): SignRecords? {
        return null
    }
}